server/integration/testsuite/src/test/java/org/infinispan/server/test/jcache/JCacheStateAwarenessIT.java

```diff
@@ -0,0 +1,184 @@
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.infinispan.arquillian.core.InfinispanResource;
+import org.infinispan.arquillian.core.RemoteInfinispanServer;
+import org.infinispan.arquillian.core.RunningServer;
+import org.infinispan.arquillian.core.WithRunningServer;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test for issues related to ISPN-6574
+ * 
+ * Test check whether JCache API is aware of caches defined in Infinispan server
+ * configuration
+ * 
+ * Three tested functionalities are not yet implemented (not part of the
+ * original ticket), fix pending in <b>ISPN-7095</b>
+ * 
+ * Calling getCache on a cache will refresh that cache inside CacheManager,
+ * making the other functionality work ok for that particular cache (hence the
+ * separate caches for each test)
+ * 
+ * @author zhostasa
+ *
+ */
+@RunWith(Arquillian.class)
+@WithRunningServer({ @RunningServer(name = "cachecontainer") })
+public class JCacheStateAwarenessIT {
+
+   private static final String enableStatisticsTestCacheName = "enableStatisticsTestCache";
+   private static final String enableManagementTestCacheName = "enableManagementTestCache";
+   private static final String getCacheNamesTestCacheName = "getCacheNamesTestCache";
+   private static final String getCacheTestCacheName = "getCacheTestCache";
+
+   private static final String testKey = "testKey";
+   private static final String testValue = "testValue";
+
+   final int managementPort = 9999;
+
+   @InfinispanResource("cachecontainer")
+   private RemoteInfinispanServer server1;
+   private RemoteCacheManager rcm1;
+
+   private CachingProvider jcacheProvider;
+   private CacheManager cacheManager;
+   private MBeanServer mBeanServer;
+
+   @Before
+   public void setUp() {
+      if (rcm1 == null) {
+         Configuration conf = new ConfigurationBuilder().addServer()
+               .host(server1.getHotrodEndpoint().getInetAddress().getHostName())
+               .port(server1.getHotrodEndpoint().getPort()).build();
+
+         rcm1 = new RemoteCacheManager(conf);
+
+         jcacheProvider = Caching.getCachingProvider("org.infinispan.jcache.remote.JCachingProvider");
+         cacheManager = jcacheProvider.getCacheManager();
+
+         mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+      }
+   }
+
+   /**
+    * Test enableStatistics command on JCache API <br>
+    * <br>
+    * Fix pending in <b>ISPN-7095</b>, needs to be uncommented when implemented
+    * 
+    */
+   // @Test
+   public void testEnableStatistics() throws Exception {
+      String mBeanName = getMBeanName(enableStatisticsTestCacheName);
```

Comment:
Should this test enable management first, or should it test that it's possible to enable only statistics?